### DISCUSSION TITLE
Poll - Strict grammar

### DISCUSSION THREAD

**Discussion by gquerret** [Riverside-Software/sonar-openedge#967] [Polls]
In an ABL procedure or class, the last `END` statement or last `.` is optional, so this kind of code can be executed without any issue:
```abl
def var zz1 as int.
def var zz2 as int.
do zz1 = 1 to 5:
  do zz2 = 1 to 10:
    message "Hello " + string(zz1 * zz2)
/* Period after message statement is optional, as well as the two END of the do statements */
```
In my opinion, that's ugly and should never have been accepted, but it works...

For multiple reasons that would be a bit too long to explain here, removing this functionality from the grammar would make things much easier when dealing with code completion (for the upcoming OpenEdge plugin for VSCode). Maintaining two versions require a significant amount of time, so we'd like to have your input on the removal of this code.

**Comment by gquerret** [comment]
In case you'd like to test how many syntax errrors you'd get, here is a snapshot of the OE plugin, based on the 2.15.0: https://ci.rssw.eu/job/sonar-openedge/job/feature%252FNoEOF/3/artifact/openedge-plugin/target/sonar-openedge-plugin-2.16.0-SNAPSHOT.jar

See next comment, it might be easier to execute the parser with a command-line rather than with a full SonarQube context.

**Comment by gquerret** [comment]
You can execute the parser on your codebase with those two JAR files: [version 2.15](https://dl.rssw.eu/parser-test-2.15.0.jar) and [version 2.16-SNAPSHOT where functionality is removed](https://dl.rssw.eu/parser-test-2.16-SNAPSHOT.jar).

Can be executed with:
```
java -jar parser-test-xxx.jar --databases xxxx --aliases xxxx --dlc xxxx --propath xxxx --src xxxx --build xxxx --encoding xxxx
```

You can reuse the same values as the ones of your code analysis. The last lines of the log output mention the files where a syntax error occured.

**Comment by cverbiest** [comment]
@gquerret I voted would accept syntax error **on condition that Sonarlint-on-the-fly will display those errors**.

Currently, sonarlint does not show anything for failing analysis.
Our developers are happy if the sonarlint on-the-fly view is empty, they do not check whether sonarlint was able to parse the code or not.

**Reply by gquerret** [reply]
Good point ! I'll check what can be done in Eclipse, and definitely keep that in my TODO list for the VSCode plugin.
